### ISSUE TITLE
SameDiff: add support for get/put ops

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/nd4j#2533]
INDArray.get(INDArrayIndex...) seems to have no analogous implementation in SameDiff. Same thing for INDArray.put(INDArrayIndex[], Number)

Conceptually, handling these at the Java level is easy, but might not be the best option, given FlatBuffer etc export.

Instead, what I propose:
1. Implement .get(INDArrayIndex...) and .put(INDArrayIndex[], INDArray) methods on SDVariable
2. For execution, use our (already implemented) Slice and StridedSlice ops, mapping INDArrayIndex to those args as necessary: https://github.com/deeplearning4j/libnd4j/search?utf8=%E2%9C%93&q=strided_slice&type=

Note: My reading of slice and strided_slice is that they are suitable for ND4J's all/interval/index args, but not for SpecifiedIndex (which, is rarely used in practice and could just be unsupported for now)

**Comment by AlexDBlack** [comment]
First: need to implement (a) slice (done here: https://github.com/deeplearning4j/nd4j/pull/2537), and (b) strided slice (requires: https://github.com/deeplearning4j/libnd4j/issues/709)

Note also for backprop: https://github.com/deeplearning4j/libnd4j/issues/710

**Comment by AlexDBlack** [comment]
Note: slice and strided_slice were added recently.
Thus only .get and .put methods (using NDArrayIndex or ideally a more suscint equivalent) remain. These are probably a lower priority right now...

**Comment by raver119** [comment]
Well, at libnd4j NDArray we have something like get implemented: NDArray::subarray(), but it gives out view. And ops do not allow that. If we'll be doing it without views, result will be equivalent to slice/strided_slice :)

So, maybe we should just provide SameDiff signature which will convert NDArrayIndex to slice/strided_slice arguments?

**Comment by AlexDBlack** [comment]
Yeah. I'm thinking maybe SDVariable.get(Idx.point(x), Idx.interval(a,b), Idx.newAxis()) or something.
Given the lack of view semantics in samediff - we probably want to keep it separate to NDArrayIndex approach. And underneath, let's just map it to the strided_slice.
